woocommerce/woocommerce-gutenberg-products-block
# Update Cart Order and Checkout Routes

This is part of #1938 and is targetting https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1957 which includes a refactor of the Store Cart API controllers.

This PR:

- Adds an `order_id` to the Cart Schema. This will contain the draft order ID during checkout, or 0 if an order has not yet been created.
- Adds a `POST cart/create-order` endpoint. This accepts an object with `billing_address`, `shipping_address` and `order_note` from the checkout and persists it to a draft order. Everything else (items, shipping, calculated totals) are taken from the **cart**. Returns an order object.
- Adds a placeholder for the `/checkout` route. We need to hash out further what is needed here for #1938 so we can process payments via gateways like the core checkout.

**Follow ups:**

- [ ] Update readme documentation
- [ ] Work on the `/checkout` endpoint - determine what it needs to accept and what it needs to return, then add processing code to the endpoint.

### How to test the changes in this Pull Request:

1. Unit tests should pass
2. Have a cart and POST to `cart/create-order`. Check the response returns a valid order with all of your items.
3. Then do a GET to `/cart`. Ensure it contains the order ID just created.
